packages/web-config-server/src/apiV1/dataBuilders/helpers/groupEvents.js

```diff
@@ -10,29 +10,71 @@ const getOrgUnits = async (models, { parentCode, type, hierarchyId }) => {
   return parentOrgUnit.getDescendantsOfType(hierarchyId, type);
 };
 
+/**
+ * Given an object of eventsByOrgUnitCode, returns the same object with the code changed to name, but
+ * disambiguated, e.g. 'Melbourne', 'Other', 'Other (1)'
+ *
+ * @param allOrgUnits
+ * @param eventsByOrgUnitCode
+ * @returns {{}}
+ * @private
+ */
+const mapOrgUnitCodeToUniqueOrgUnitName = (allOrgUnits, eventsByOrgUnitCode) => {
+  const orgUnitsByCode = {};
+
+  for (const orgUnitCode of Object.keys(eventsByOrgUnitCode)) {
+    orgUnitsByCode[orgUnitCode] = allOrgUnits.find(orgUnit => orgUnit.code === orgUnitCode);
+  }
+
+  const eventsByUniqueOrgUnitName = {};
+
+  for (const orgUnitCode of Object.keys(eventsByOrgUnitCode)) {
+    const orgUnit = orgUnitsByCode[orgUnitCode];
+
+    const isNameUnique =
+      Object.values(orgUnitsByCode)
+        .filter(otherOrgUnit => otherOrgUnit.code !== orgUnitCode)
+        .filter(otherOrgUnit => otherOrgUnit.name === orgUnit.name).length === 0;
+
+    if (isNameUnique) {
+      eventsByUniqueOrgUnitName[orgUnit.name] = eventsByOrgUnitCode[orgUnitCode];
+    } else {
+      const uniqueName = `${orgUnit.name} (${orgUnitCode})`;
+      eventsByUniqueOrgUnitName[uniqueName] = eventsByOrgUnitCode[orgUnitCode];
+    }
```

Comment:
This could be shortened to two lines if you wanted:
```suggestion
      const uniqueName = isNameUnique ? orgUnit.name : `${orgUnit.name} (${orgUnitCode})`;
      eventsByUniqueOrgUnitName[uniqueName] = eventsByOrgUnitCode[orgUnitCode];
```